packages/ember-data/lib/serializers/json-api-serializer.js

```diff
@@ -0,0 +1,135 @@
+/**
+  @module ember-data
+*/
+
+import JSONSerializer from 'ember-data/serializers/json-serializer';
+
+/**
+  @class JSONAPISerializer
+  @namespace DS
+  @extends DS.JSONSerializer
+*/
+export default JSONSerializer.extend({
+
+  /*
+    @method _normalizeResponse
+    @param {DS.Store} store
+    @param {DS.Model} primaryModelClass
+    @param {Object} payload
+    @param {String|Number} id
+    @param {String} requestType
+    @param {Boolean} isSingle
+    @return {Object} JSON-API Document
+    @private
+  */
+  _normalizeResponse: function(store, primaryModelClass, payload, id, requestType, isSingle) {
+    return payload;
+  },
+
+  /**
+    @method serialize
+    @param {DS.Snapshot} snapshot
+    @param {Object} options
+    @return {Object} json
+  */
+  serialize: function(snapshot, options) {
+    let data = this._super(...arguments);
+    data.type = snapshot.modelName;
+    return { data };
+  },
+
+  /**
+   @method serializeAttribute
+   @param {DS.Snapshot} snapshot
+   @param {Object} json
+   @param {String} key
+   @param {Object} attribute
+  */
+  serializeAttribute: function(snapshot, json, key, attribute) {
+    var type = attribute.type;
+
+    if (this._canSerialize(key)) {
+      json.attributes = json.attributes || {};
+
+      var value = snapshot.attr(key);
+      if (type) {
+        var transform = this.transformFor(type);
+        value = transform.serialize(value);
+      }
+
+      var payloadKey =  this._getMappedKey(key);
+      if (payloadKey === key && this.keyForAttribute) {
```

Comment:
Do we need to check `this.keyForAttribute` if we know the `JSONSerializer` has it defined?